src/actions/sources.js

```diff
@@ -49,11 +49,23 @@ import type { ThunkArgs } from "./types";
 
 // If a request has been made to show this source, go ahead and
 // select it.
-function checkSelectedSource(source) {
+function checkSelectedSource(source: Source) {
   return async ({ dispatch, getState }: ThunkArgs) => {
     const pendingLocation = getPendingSelectedLocation(getState());
 
-    if (pendingLocation && !!source.url && pendingLocation.url === source.url) {
+    if (!pendingLocation || !pendingLocation.hasOwnProperty("url")) {
+      return;
+    }
+
+    const { url: pendingURL } = pendingLocation;
+    const formattedPendingURL = pendingURL.replace(/:formatted$/, "");
+    if (!!source.url && formattedPendingURL === source.url) {
+      const isPendingLocationPretty = pendingURL.endsWith(":formatted");
+
+      if (isPendingLocationPretty) {
+        return await dispatch(togglePrettyPrint(source.id));
```

Comment:
do we need to do `selectLocation` if we do this?

Reply:
Nope, works out-of-the box